### DISCUSSION TITLE
Is there a way to control the contents of the Installing and Example tabs on a pub.dev package's page?

### DISCUSSION THREAD

**Discussion by nfriedly** [dart-lang/pub-dev#8226] [Q&A] [answered]
I recently published a [package](https://pub.dev/packages/fullstory_flutter) to pub.dev. Overall I'm pretty happy with the generated pages for it, but there are a couple of small tweaks I'd like to make:

* On the Installing page, the instructions say to import every top-level file in my lib/ directory, when really only one is needed because it re-exports the contents of the others. Can I change that without changing my codebase? (It does not instruct users to import my lib/src/ files, at least.)

* On the Example page, it's showing the `main.dart` file from my example app, which is basically just imports and "fluff". Nothing in it is particularly relevant for using my library. I'd ideally like to show the contents of half a dozen or so other files. Is this possible?

I know I could rearrange the source code of my library to put everything in a single file, but that feels like the wrong solution.

**Comment by isoos** [comment, marked as answer]
> On the Installing page, the instructions say to import every top-level file in my lib/ directory, when really only one is needed because it re-exports the contents of the others. Can I change that without changing my codebase? (It does not instruct users to import my lib/src/ files, at least.)

If you have a `lib/<package_name>.dart`, we treat it as the primary library, and we will list only that (and we use only that as an entry point during static analysis). `lib/src/` is treated as package internals and are not exposed here.

> On the Example page, it's showing the main.dart file from my example app, which is basically just imports and "fluff". Nothing in it is particularly relevant for using my library. I'd ideally like to show the contents of half a dozen or so other files. Is this possible?

You can also put `README.md` or `example.md` in the example directory with code blocks and further links. https://dart.dev/tools/pub/package-layout#examples

**Comment by nfriedly** [comment]
> If you have a lib/<package_name>.dart, we treat it as the primary library, and we will list only that (and we use only that as an entry point during static analysis).

Ok, that's good to know, thanks!

> You can also put README.md or example.md in the example directory with code blocks and further links. https://dart.dev/tools/pub/package-layout#examples

Aah, I see. I have a `README.md`, but it comes lower down the list than `example/lib/main.dart`. I can make an `example.md`, though. Thanks again!

**Comment by nfriedly** [comment]
I put together a script to generate an `example.md` that's essentially just the contents of all the files in my `example/lib/` directory + a table of contents at the top: https://github.com/fullstorydev/fullstory-flutter/blob/cb22dc8aecd86780c95f84d37f898b2d41e4f301/generate_example_md.sh

I initially tried just running it in CI before publish, with the output .gitignore'd, but apparently `flutter pub publish` also ignores that. So I checked in the file also, and now it's showing up on my example page!
